packages/fxa-settings/src/lib/sensitive-data-client.ts

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+export class SensitiveDataClient {
+  private sensitiveData: { [key: string]: string } = {};
+  private key: CryptoKey | null = null;
+  private iv: Uint8Array | undefined;
+
+  constructor() {
+    this.sensitiveData = {};
+  }
+
+  async init() {
+    this.key = await window.crypto.subtle.generateKey(
+      {
+        name: 'AES-GCM',
+        length: 256,
+      },
+      true,
+      ['encrypt', 'decrypt']
+    );
+    this.iv = window.crypto.getRandomValues(new Uint8Array(12));
```

Comment:
I don't think encryption is necessary.  Is the idea that the in-process values are encrypted in RAM?  If someone has that kind of access, they could read the values before the values enter or after the values exit this module anyway.

Reply:
I was on the fence about encryption, but remember Wil mention not storing anything in plaintext. Maybe it is overkill but perhaps alright?

Reply:
I thought "storing" meant persisting the value onto something that outlives the lifespan of the page.

Reply:
I agree, I think storing would be writing the value to durable store (ie session storage, local storage, local db etc..)

Reply:
I'll update and remove encryption